### ISSUE TITLE
(Feature request) Shortcut to create sub-note and include it

### ISSUE THREAD

**Issue by salmund** [zadam/trilium#3542]
### Describe feature

It would be very useful to have a shortcut to create a subnote from the selected text of the current note and include this subnote in the current note.

Often I have a piece of text in a note that I want to make more easily accessible by creating a sub-note from that text while keeping the sub-note text in the current note (thus including the sub-note).

The most time consuming approach is to write a title and then do a "Cut & paste selection to sub-note" and then include the sub-note in the current note.

This step could simply be a "Create sub-note from select text and include it in note" function that could be called directly with a keyboard shortcut, which would be even more efficient.

**Comment by contributor** [comment]
![image](https://user-images.githubusercontent.com/598873/213573985-9ccb9790-0697-4475-9e65-d5c244a26350.png)

**Comment by salmund** [comment]
> ![image](https://user-images.githubusercontent.com/598873/213573985-9ccb9790-0697-4475-9e65-d5c244a26350.png)

This does not include the sub-note in the original starting note, it just creates a sub-note, which corresponds to 1 step out of 2, the second step being to include it (automatically).

I'm sure it can be done with the Script API, but I don't know how to do it.

**Comment by contributor** [comment]
I see. How do you include it manually? Using this dialog?
![image](https://user-images.githubusercontent.com/598873/213704637-16238b74-9a40-40e7-a770-217dbe973501.png)

What option the shortcut will use then? This feature will require additional setting for that I think. Or apply "last used option" autmoatically, but sometimes wrong option can be applied (and there seem no way to change that after note is included)

**Comment by salmund** [comment]
Yes, I use this shortcut actually :

![image](https://user-images.githubusercontent.com/105465034/213794444-b4789657-9cbb-48fb-9ace-88921ccaac74.png)

**Comment by contributor** [comment]
I agree, the feature to do these two actions in one go would be very useful and productive

**Comment by shutaozhenzhen** [comment]
https://github.com/zadam/trilium/discussions/3030

**Comment by salmund** [comment]
I did the job!

[https://github.com/salmund/trilium_better_include/tree/main](url)

![Animation](https://user-images.githubusercontent.com/105465034/222968795-0a488e29-57f6-4add-a356-521cadcc31c9.gif)

the code is a bit messy, a lot of optimization can be done I think, but I'm really, really lazy and it does the job, so that's it!

**Comment by eliandoran** [comment]
Looks nice but I don't think it's better we kept it as a separate plugin like https://github.com/salmund/trilium_better_include/.
